### PR TITLE
Rust: Add context to the traits

### PR DESCRIPTION
**Pull request by dorezyuk**
Extends the traits with the context struct

### CODE REVIEW

**Comment by SirVer** [comment]
I see the need for having the information about the interface sometimes. The generated code has it always, right? How about instead of adding a new trait we bundle ths information into the ModulePublisher, i.e. make a `struct Context { implementation_id: str, publishers: ModulePublishers }`? I find the ergonomics of an extended trait a bit burdensom also, it seems that it does not extend well if we find that there is other such information.

File: everestrs/everestrs-build/jinja/module.jinja2 [hunk 1]
```diff
@@ -43,6 +43,12 @@ pub(crate) struct Module {
     publisher: ModulePublisher,
 }
 
+/// The context structure.
+pub struct Context<'a> {
+    pub publisher: &'a ModulePublisher,
+    pub name: &'a str,
```
Comment (SirVer) [inline everestrs/everestrs-build/jinja/module.jinja2:49]: should this not be `implementation_id`. I am not so clear on EVerest nomenclatures.

**Review by SirVer — APPROVED** [review]
Please fix the PR description to represent the new design, otherwise lgtm.

File: everestrs/everestrs-build/jinja/module.jinja2 (hunk 1 above)
Comment (dorezyuk) [reply everestrs/everestrs-build/jinja/module.jinja2:49]: i'm also not sure

**Comment by SirVer** [comment]
Please still fix the MR headline. This is no longer about extended traits.

**Pull request merged**